relayer/provider/cosmos/provider.go

```diff
@@ -1709,13 +1697,16 @@ func (cc *CosmosProvider) SendMessages(ctx context.Context, msgs []provider.Rela
 		return nil, false, err
 	}
 
-	// Parse events and build a map where the key is event.Type+"."+attribute.Key
-	events := make(map[string]string, 1)
+	// Parse events from the tx response.
+	var events []*provider.RelayerEvent
 	for _, logs := range res.Logs {
-		for _, ev := range logs.Events {
-			for _, attr := range ev.Attributes {
-				key := ev.Type + "." + attr.Key
-				events[key] = attr.Value
+		for _, event := range logs.Events {
+			for _, attr := range event.Attributes {
+				events = append(events, &provider.RelayerEvent{
```

Comment:
Should this use a variation of the updated `parseEventsFromResponseDeliverTx` function? It's only a couple lines so maybe it doesn't matter.

Reply:
Yeah it's more readable to tuck this into it's own function, i'll make those changes as well.